File: app/src/androidTest/java/com/cvv/fanstaticapps/randomticker/SetProgressAction.java
package com.cvv.fanstaticapps.randomticker;

import android.view.View;
import android.widget.SeekBar;

import org.hamcrest.Matcher;

import androidx.test.espresso.UiController;
import androidx.test.espresso.ViewAction;

import static androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static org.hamcrest.Matchers.allOf;

/**
 * Created by Carla
 * Date: 05/02/2018
 * Project: RandomTicker
 */
public class SetProgressAction implements ViewAction {
    private final int progress;

    public SetProgressAction(int progress) {
        this.progress = progress;
    }

    @Override
    public Matcher<View> getConstraints() {
        return allOf(isDisplayed(), isAssignableFrom(SeekBar.class));
    }

    @Override
    public String getDescription() {
        return "set progress";
    }

    @Override
    public void perform(UiController uiController, View view) {
        ((SeekBar) view).setProgress(progress);
    }

    public static SetProgressAction setProgress(int progress) {
        return new SetProgressAction(progress);
    }
}

File: app/src/androidTest/java/com/cvv/fanstaticapps/randomticker/SetProgressAction.kt
package com.cvv.fanstaticapps.randomticker

import android.view.View
import android.widget.SeekBar
import androidx.test.espresso.UiController
import androidx.test.espresso.ViewAction
import androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import org.hamcrest.Matcher
import org.hamcrest.Matchers.allOf

/**
 * Created by carvaq
 * Date: 05/02/2018
 * Project: RandomTicker
 */
class SetProgressAction(private val progress: Int) : ViewAction {

    override fun getConstraints(): Matcher<View> {
        return allOf(isDisplayed(), isAssignableFrom(SeekBar::class.java))
    }

    override fun getDescription(): String {
        return "set progress"
    }

    override fun perform(uiController: UiController, view: View) {
        (view as SeekBar).progress = progress
    }

    companion object {

        fun setProgress(progress: Int): SetProgressAction {
            return SetProgressAction(progress)
        }
    }
}
